src/main/java/frc/robot/subsystems/shooter/Shooter.java

```diff
@@ -0,0 +1,138 @@
+package frc.robot.subsystems.shooter;
+
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonFX;
+import edu.wpi.first.math.Nat;
+import edu.wpi.first.math.VecBuilder;
+import edu.wpi.first.math.controller.LinearQuadraticRegulator;
+import edu.wpi.first.math.estimator.KalmanFilter;
+import edu.wpi.first.math.numbers.N1;
+import edu.wpi.first.math.system.LinearSystem;
+import edu.wpi.first.math.system.LinearSystemLoop;
+import edu.wpi.first.math.system.plant.DCMotor;
+import edu.wpi.first.math.system.plant.LinearSystemId;
+import edu.wpi.first.wpilibj.Timer;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Constants;
+import frc.robot.Robot;
+import frc.robot.subsystems.UnitModel;
+import frc.robot.utils.Utils;
+
+import static frc.robot.Constants.*;
+import static frc.robot.Constants.Shooter.*;
+import static frc.robot.Ports.Shooter.*;
+
+public class Shooter extends SubsystemBase {
+    private static Shooter INSTANCE;
+    private final UnitModel unitModel = new UnitModel(TICKS_PER_RADIAN);
+    private final WPI_TalonFX mainMotor = new WPI_TalonFX(MAIN_MOTOR);
+    private final LinearSystemLoop<N1, N1, N1> linearSystemLoop;
+    private double currentTime = 0;
+    private double lastTime = 0;
+
+    /**
+     * Constructor.
+     */
+    private Shooter() {
+        CONFIGURATION.enableOptimizations = true;
```

Comment:
If you only use and configure the motor, create the CONFIGURATION in the constructor